convert/convert.go

```diff
@@ -129,10 +148,13 @@ func (c *Converter) reset(size int64) {
 	c.offset = 0
 }
 
-// Convert copy size bytes from io.ReaderAt to io.WriterAt. Unallocated areas or
-// areas full of zeros in the source are keep unallocated in the destination.
-// The destination must be new empty or full of zeroes.
-func (c *Converter) Convert(wa io.WriterAt, ra io.ReaderAt, size int64) error {
+// Convert copy size bytes from image to io.WriterAt. Unallocated extents in the
+// source image or read data which is all zeros are converted to unallocated
+// byte range in the target image. The target image must be new empty file or a
+// file full of zeroes. To get a sparse target image, the image must be a new
+// empty file, since Convert does not punch holes for zero ranges even if the
+// underlying file system supports hole punching.
+func (c *Converter) Convert(wa io.WriterAt, img image.Image, size int64, progress Updater) error {
```

Comment:
`progress Updater` could be provided in `Options`?
Then we would not have to change the function signature?

Reply:
It is possible, but even if we do we accept now an image instead of a ReaderAt and size.

Also progress scope is one convert, while convert instance can convert many images using the same options, so it is better to accept progress per convert operation.

The options are not something you change per image, it may change depending on the machine (e.g number of cores).

Reply:
There other options that we may want to add in the future that may be per image options, so moving the options to Convert makes sense anyway.